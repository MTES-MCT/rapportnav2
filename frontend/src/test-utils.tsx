import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
