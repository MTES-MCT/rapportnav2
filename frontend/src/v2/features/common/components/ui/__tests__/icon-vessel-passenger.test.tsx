import { render } from '../../../../../../test-utils'
import IconVesselPassenger from '../icon-vessel-passenger'

describe('IconVesselPassenger', () => {
  it('should match the snapshot', () => {
    const wrapper = render(<IconVesselPassenger />)
    expect(wrapper).toMatchSnapshot()
  })
})
