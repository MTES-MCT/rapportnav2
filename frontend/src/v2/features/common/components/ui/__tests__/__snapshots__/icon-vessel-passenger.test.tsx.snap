// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IconVesselPassenger > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <img
        alt="Navire passagers"
        src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='75'%20height='36'%20viewBox='0%200%2075%2036'%3e%3cdefs%3e%3cclipPath%20id='clip-path'%3e%3crect%20id='Rectangle_6791'%20width='71.157'%20height='27.834'%20fill='%233b4559'/%3e%3c/clipPath%3e%3cclipPath%20id='clip-Plan_de_travail_3'%3e%3crect%20width='75'%20height='36'/%3e%3c/clipPath%3e%3c/defs%3e%3cg%20id='Plan_de_travail_3'%20clip-path='url(%23clip-Plan_de_travail_3)'%3e%3crect%20width='75'%20height='36'%20fill='%23fff'/%3e%3cg%20id='Groupe_5544'%20transform='translate(2%204)'%3e%3cg%20id='Groupe_5530'%20clip-path='url(%23clip-path)'%3e%3cpath%20id='Tracé_1835'%20d='M16.317,14.615H14.63a1.688,1.688,0,1,1,0-3.375h1.687a1.688,1.688,0,0,1,0,3.375'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1836'%20d='M11.676,7.584H10.551a1.266,1.266,0,0,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1837'%20d='M16.457,7.584H15.332a1.266,1.266,0,0,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1838'%20d='M30.238,7.584H29.113a1.266,1.266,0,1,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1839'%20d='M35.019,7.584H33.894a1.266,1.266,0,0,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1840'%20d='M35.722,14.615H33.191a1.688,1.688,0,1,1,0-3.375h2.531a1.688,1.688,0,0,1,0,3.375'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1841'%20d='M42.471,14.615H39.94a1.688,1.688,0,1,1,0-3.375h2.531a1.688,1.688,0,0,1,0,3.375'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1842'%20d='M50.283,14.615H45.627A.625.625,0,0,1,45,13.99V11.865a.625.625,0,0,1,.625-.625H48.6c.345,0,.906.28.906.625l1.125,2.125c0,.345,0,.625-.344.625'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1843'%20d='M11.473,14.615H6.255a.625.625,0,0,1-.625-.625l.562-2.125a.625.625,0,0,1,.625-.625h4.656a.625.625,0,0,1,.625.625V13.99a.625.625,0,0,1-.625.625'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1844'%20d='M54.666,16.958,50.114,9.083a.375.375,0,0,0-.325-.187H37.929l1.708-5.132a.376.376,0,0,0-.356-.493H29.367a.35.35,0,0,0-.067-.337L26.718.122A.373.373,0,0,0,26.441,0h-16.2a.374.374,0,0,0-.375.375V3.188a.34.34,0,0,0,.017.083H7.478a.375.375,0,0,0-.375.375V8.9H4.786a.375.375,0,0,0-.366.292L2.639,17.063a.375.375,0,0,0,.283.448.4.4,0,0,0,.083.009.376.376,0,0,0,.366-.292L5.086,9.646H19.317v7.5a.375.375,0,0,0,.375.375h6.187a.375.375,0,0,0,.375-.375v-7.5H49.573l4.443,7.687a.375.375,0,0,0,.65-.375M28.591,3.271h-2.9V.75h.586ZM13.317.75h2.344V3.271H13.3a.4.4,0,0,0,.017-.083Zm3.094,0h2.344V3.271H16.411ZM19.5.75h2.343V3.271H19.5ZM22.6.75h2.344V3.271H22.6ZM10.619,3.188V.75h1.948V3.188a.34.34,0,0,0,.017.083H10.6a.34.34,0,0,0,.017-.083M25.5,16.77H20.067v-9.7A1.93,1.93,0,0,1,22,5.146h1.584A1.927,1.927,0,0,1,25.5,7.07ZM23.579,4.4H22a2.681,2.681,0,0,0-2.678,2.678V8.9H7.853V4.021H38.761L37.139,8.9H26.254V7.07A2.677,2.677,0,0,0,23.579,4.4'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1845'%20d='M0,17.145S-.276,23.9,3.38,26.426c0,0,.843,1.406,6.468,1.406h48.4s7.846.281,11.221-6.75l1.687-3.937Zm11.531,2.109a1.048,1.048,0,1,1-.443,2,1.3,1.3,0,0,0-.6-.964,1.048,1.048,0,0,1,1.047-1.032M9.522,20.617a1.047,1.047,0,1,1-1.048,1.048,1.048,1.048,0,0,1,1.048-1.048M7.4,19.255A1.048,1.048,0,0,1,8.45,20.3c0,.02,0,.039,0,.059a1.292,1.292,0,0,0-.487.827A1.047,1.047,0,1,1,7.4,19.255m-.2,4.872v.328a6.3,6.3,0,0,1-1.511-.284l-.075-.023,0-.017V23.044A1.343,1.343,0,0,1,6.958,21.7h.889c.033,0,.065,0,.1.005a1.3,1.3,0,0,0,.4.895A1.593,1.593,0,0,0,7.2,24.127m4.1,1.085H11.3l-.07.036a3.745,3.745,0,0,1-1.593.3,6.118,6.118,0,0,1-1.823-.294l-.075-.024,0-.017V24.125a1.343,1.343,0,0,1,1.341-1.341h.889a1.342,1.342,0,0,1,1.341,1.341Zm2.015-1.081h-.007l-.07.035a3.6,3.6,0,0,1-1.406.3v-.335A1.593,1.593,0,0,0,10.7,22.6a1.3,1.3,0,0,0,.4-.9h.882a1.343,1.343,0,0,1,1.341,1.342Z'%20fill='%233b4559'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
      />
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <img
      alt="Navire passagers"
      src="data:image/svg+xml,%3csvg%20xmlns='http://www.w3.org/2000/svg'%20width='75'%20height='36'%20viewBox='0%200%2075%2036'%3e%3cdefs%3e%3cclipPath%20id='clip-path'%3e%3crect%20id='Rectangle_6791'%20width='71.157'%20height='27.834'%20fill='%233b4559'/%3e%3c/clipPath%3e%3cclipPath%20id='clip-Plan_de_travail_3'%3e%3crect%20width='75'%20height='36'/%3e%3c/clipPath%3e%3c/defs%3e%3cg%20id='Plan_de_travail_3'%20clip-path='url(%23clip-Plan_de_travail_3)'%3e%3crect%20width='75'%20height='36'%20fill='%23fff'/%3e%3cg%20id='Groupe_5544'%20transform='translate(2%204)'%3e%3cg%20id='Groupe_5530'%20clip-path='url(%23clip-path)'%3e%3cpath%20id='Tracé_1835'%20d='M16.317,14.615H14.63a1.688,1.688,0,1,1,0-3.375h1.687a1.688,1.688,0,0,1,0,3.375'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1836'%20d='M11.676,7.584H10.551a1.266,1.266,0,0,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1837'%20d='M16.457,7.584H15.332a1.266,1.266,0,0,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1838'%20d='M30.238,7.584H29.113a1.266,1.266,0,1,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1839'%20d='M35.019,7.584H33.894a1.266,1.266,0,0,1,0-2.531h1.125a1.266,1.266,0,1,1,0,2.531'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1840'%20d='M35.722,14.615H33.191a1.688,1.688,0,1,1,0-3.375h2.531a1.688,1.688,0,0,1,0,3.375'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1841'%20d='M42.471,14.615H39.94a1.688,1.688,0,1,1,0-3.375h2.531a1.688,1.688,0,0,1,0,3.375'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1842'%20d='M50.283,14.615H45.627A.625.625,0,0,1,45,13.99V11.865a.625.625,0,0,1,.625-.625H48.6c.345,0,.906.28.906.625l1.125,2.125c0,.345,0,.625-.344.625'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1843'%20d='M11.473,14.615H6.255a.625.625,0,0,1-.625-.625l.562-2.125a.625.625,0,0,1,.625-.625h4.656a.625.625,0,0,1,.625.625V13.99a.625.625,0,0,1-.625.625'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1844'%20d='M54.666,16.958,50.114,9.083a.375.375,0,0,0-.325-.187H37.929l1.708-5.132a.376.376,0,0,0-.356-.493H29.367a.35.35,0,0,0-.067-.337L26.718.122A.373.373,0,0,0,26.441,0h-16.2a.374.374,0,0,0-.375.375V3.188a.34.34,0,0,0,.017.083H7.478a.375.375,0,0,0-.375.375V8.9H4.786a.375.375,0,0,0-.366.292L2.639,17.063a.375.375,0,0,0,.283.448.4.4,0,0,0,.083.009.376.376,0,0,0,.366-.292L5.086,9.646H19.317v7.5a.375.375,0,0,0,.375.375h6.187a.375.375,0,0,0,.375-.375v-7.5H49.573l4.443,7.687a.375.375,0,0,0,.65-.375M28.591,3.271h-2.9V.75h.586ZM13.317.75h2.344V3.271H13.3a.4.4,0,0,0,.017-.083Zm3.094,0h2.344V3.271H16.411ZM19.5.75h2.343V3.271H19.5ZM22.6.75h2.344V3.271H22.6ZM10.619,3.188V.75h1.948V3.188a.34.34,0,0,0,.017.083H10.6a.34.34,0,0,0,.017-.083M25.5,16.77H20.067v-9.7A1.93,1.93,0,0,1,22,5.146h1.584A1.927,1.927,0,0,1,25.5,7.07ZM23.579,4.4H22a2.681,2.681,0,0,0-2.678,2.678V8.9H7.853V4.021H38.761L37.139,8.9H26.254V7.07A2.677,2.677,0,0,0,23.579,4.4'%20fill='%233b4559'/%3e%3cpath%20id='Tracé_1845'%20d='M0,17.145S-.276,23.9,3.38,26.426c0,0,.843,1.406,6.468,1.406h48.4s7.846.281,11.221-6.75l1.687-3.937Zm11.531,2.109a1.048,1.048,0,1,1-.443,2,1.3,1.3,0,0,0-.6-.964,1.048,1.048,0,0,1,1.047-1.032M9.522,20.617a1.047,1.047,0,1,1-1.048,1.048,1.048,1.048,0,0,1,1.048-1.048M7.4,19.255A1.048,1.048,0,0,1,8.45,20.3c0,.02,0,.039,0,.059a1.292,1.292,0,0,0-.487.827A1.047,1.047,0,1,1,7.4,19.255m-.2,4.872v.328a6.3,6.3,0,0,1-1.511-.284l-.075-.023,0-.017V23.044A1.343,1.343,0,0,1,6.958,21.7h.889c.033,0,.065,0,.1.005a1.3,1.3,0,0,0,.4.895A1.593,1.593,0,0,0,7.2,24.127m4.1,1.085H11.3l-.07.036a3.745,3.745,0,0,1-1.593.3,6.118,6.118,0,0,1-1.823-.294l-.075-.024,0-.017V24.125a1.343,1.343,0,0,1,1.341-1.341h.889a1.342,1.342,0,0,1,1.341,1.341Zm2.015-1.081h-.007l-.07.035a3.6,3.6,0,0,1-1.406.3v-.335A1.593,1.593,0,0,0,10.7,22.6a1.3,1.3,0,0,0,.4-.9h.882a1.343,1.343,0,0,1,1.341,1.342Z'%20fill='%233b4559'/%3e%3c/g%3e%3c/g%3e%3c/g%3e%3c/svg%3e"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
